app.py

```diff
@@ -98,16 +106,19 @@ def main():
     cols = list(dataset.columns)
     st.write(f"This dataset has ${dataset.shape[0]}$ records and {cols} fields")
 
-    st.header("Additional features")
+    "Here's a sample sample of the raw dataset"
+    st.table(df_rating.head())
+    f"Total of ${df_rating.shape[0]}$ records"
 
-    st.subheader("Average ratings and user count")
+    st.header("Additional features")
 
-    """
-        Now that we have a basic data set with one hot encoded genres, the next step
-        is to add additional information from the `ratings.csv` and optionally
-        `genome-*.csv`.
-    """
+    st.subheader("Average rating and user count")
+    average_rating = df_rating["rating"].mean(axis=0)
+    f"`Average Rating Overall: {average_rating}`"
 
+    rows = list(df_rating.index.values)
+    distinct_rows = set(rows)
+    f"`User count: {len(distinct_rows)}`"
```

Comment:
You have to group by movieId and then calculate the mean, something like
`df_rating.groupby('movieId').mean()`. I will update this and push later